ophyd/PreDefinedPositions.py

```diff
@@ -0,0 +1,321 @@
+        is used identically to the read attribute function for a standard device and therefore can 
+        be used in the baseline.
+        
+        Parameters
+        ----------
+        read_dict: ordered dictionary, ouput
+            The output dictionary that matches the standard output for a Device.
+        '''
+
+        out_dict = collections.OrderedDict()
+        out_dict[self.name+'_location'] = {'timestamp':time.time(),'value':self.position }
+
+        read_dict = super().read()
+        read_dict.update(out_dict)
+        
+        return read_dict
+
+    
+    
+    def describe(self):
+        '''
+        An attribute that returns the current 'location' description of the output data as an 
+        ordered dictionary. This is used identically to the describe attribute function for a 
+        standard device and therefore can be used in the baseline.
+        
+        Parameters
+        ----------
+        describe_dict: ordered dictionary, ouput
+            The output dictionary that matches the standard output for a Device.
+        '''
+
+        out_dict = collections.OrderedDict()
+        out_dict[self.name+'_location'] = {'dtype': 'string',
+               'lower_ctrl_limit': None,
+               'precision': None,
+               'shape': [],
+               'source': 'None',
+               'units': None,
+               'upper_ctrl_limit': None}
+              
+        describe_dict = super().describe()
+        describe_dict.update(out_dict)
+        
+        return describe_dict
+
+
+    def find_path(self,from_location=None,to_location=None):
+        '''
+        Find the shortest path from 'from_location' to 'to_location' passing only thorugh the 
+        neighbours for each location defiend by the dictionary 'neighbours'. Returns an empty list 
+        if no path found otherwise returns a list of 'locations' that define the path. If to_location 
+        is None then it returns a dictionary showing the shortest path to all possible locations. If
+        from_location is None it returns a dictionary showing the shortest path from all locations to
+        the current location. If both are None it returns a dictioanry of dictionaries. If 
+        from_location is 'current_location' the starting point is changed to the current location.
+        Parameters
+        ---------
+        from_location: string
+            The name of the starting location required for the path.
+        to_location: string
+            The name of the ending location required for the path.
+        path_list: list, output
+            A list locations indicating the path to take to reach the required position.
+        '''
+
+
+        if from_location is not 'current_location':
+            path_list = nx.shortest_path(self.nxGraph,source=from_location,target=to_location)
+        elif isinstance(self.status_list,str) and self.neighbours is not None:
+            path_list=(self.status_list)
```

Comment:
do you need parentheses here? Are you converting a string value to a list? Then `[]` should be used.